File: src/main/java/net/numa08/AkkaDanyo.java
package net.numa08;

import net.numa08.provider.DejikoProvider;
import net.numa08.provider.DejikoProviderFactory;

public class AkkaDanyo {

    public static void main(String... args) {
        try {
            if (args.length == 0) {
                throw new IllegalArgumentException("Invalid Number of Argument!!!!");
            }
            final DejikoProvider provider = DejikoProviderFactory.providerByName(args[0]);
            final int retval = provider.downLoadDejikoAt(".");
            System.out.println(retval);
        } catch (Exception e) {
            System.err.println(e.getMessage());
            System.exit(-1);
        }
    }
}

File: src/main/scala/net/numa08/AkkaDanyo.scala
package net.numa08

import net.numa08.scala.provider.DejikoProviderFactory

object AkkaDanyo {

  def main(args : Array[String]): Unit = {
    args.headOption match {
      case Some(service) => {
        val provider = DejikoProviderFactory.providerByName(service)
        provider.downloadImageAt("dejiko_scala")
      }
      case _ => {
        println(
          """usage: run [provider]
            |download dejiko-chan illust by [provider]
            |
            |provider -- A service which provide illusts. You can choice pixiv
          """.stripMargin)
      }
    }
  }
}
